model/task/task.go

```diff
@@ -1476,15 +1473,44 @@ func (t *Task) SetStepbackDepth(stepbackDepth int) error {
 		})
 }
 
-// output returns the populated task output metadata when appropriate. This is
-// needed to support backwards compatibility for tasks that do not have the
-// output metadata saved in the database.
-func (t *Task) output() *taskoutput.TaskOutput {
+// initializeTaskOutputInfo returns the task output information with the most
+// up-to-date configuration for the task run. Returns false if the task will
+// never have output. This function should only be used to set the task output
+// field upon task dispatch.
+func (t *Task) initializeTaskOutputInfo(env evergreen.Environment) (*taskoutput.TaskOutput, bool) {
 	if t.DisplayOnly {
-		return nil
+		return nil, false
+	}
+
+	taskID := t.Id
+	if t.Archived {
+		taskID = t.OldTaskId
 	}
 
-	return &taskoutput.TaskOutput{}
+	return taskoutput.InitializeTaskOutput(env, taskoutput.TaskOptions{
+		ProjectID: t.Project,
+		TaskID:    taskID,
+		Execution: t.Execution,
+	}), true
+}
+
+// getTaskOutputSafe returns an instantiation of the task output interface and
+// whether it is safe to fetch task output data. This function should always
+// be called to access task output data.
+func (t *Task) getTaskOutputSafe() (*taskoutput.TaskOutput, bool) {
```

Comment:
I guess if this is going to return a bool and not an error, it's not really that much better than just passing the ponter back and letting it be nil/non-nil. That being said, I'm also fine if you want to leave it like this (to avoid unintentional nil pointer derefs).

Reply:
Yeah I think it is more clear in the case, similar to the map convention of returning a bool if the key does not exist.